db/local_db/start_database.go

```diff
@@ -244,26 +252,122 @@ func startPostgresProcess(port int, listen StartListenType, invoker constants.In
 	}
 	err = runningInfo.Save()
 	if err != nil {
-		postgresCmd.Process.Kill()
+		killOnStartFail(postgresCmd)
 		return err
 	}
 
-	err = postgresCmd.Process.Release()
+	connection, err := createDbClient("postgres", constants.DatabaseSuperUser)
 	if err != nil {
-		postgresCmd.Process.Kill()
+		dumpDiagnosticData()
+		killOnStartFail(postgresCmd)
 		return err
 	}
+	connection.Close()
 
-	connection, err := createDbClient("postgres", constants.DatabaseSuperUser)
+	err = postgresCmd.Process.Release()
 	if err != nil {
-		postgresCmd.Process.Kill()
+		killOnStartFail(postgresCmd)
 		return err
 	}
-	connection.Close()
 
 	return nil
 }
 
+func killOnStartFail(cmd *exec.Cmd) {
+	// get the process
+	p, _ := psutils.NewProcess(int32(cmd.Process.Pid))
+	// kill it
+	doThreeStepPostgresExit(p)
+}
+
+func dumpDiagnosticData() {
+	dumpingDone := make(chan bool, 1)
+	sp := display.StartSpinnerAfterDelay("This shouldn't have happened. Collecting diagnostic data", constants.SpinnerShowTimeout, dumpingDone)
+	dump := DiagnosticDumpData{}
+
+	wg := sync.WaitGroup{}
+	wg.Add(5)
+
+	go dumpProcesses(&dump, &wg)
+	go dumpPortUsage(&dump, &wg)
+	go dumpInstalled(&dump, &wg)
+	go dumpState(&dump, &wg)
+
+	// We don't want to lose the call stack. DON'T run this is a goroutine
+	dump.CallStack = string(debug.Stack())
+	dump.Version = version.String()
+	wg.Done()
+
+	// wait for the whole thing to get filled up
+	wg.Wait()
+
+	dumpFileLocation := filepath.Join(getDatabaseLogDirectory(), fmt.Sprintf("dump-%s.dump", time.Now().Format("20060102150405Z")))
+	dumpFile, _ := os.Create(dumpFileLocation)
+
+	// use an YAML encoder, since the user may need to inspect and edit this file
+	// a YAML is easier to work with
+	ymlEncoder := yaml.NewEncoder(dumpFile)
+	ymlEncoder.Encode(dump)
+	ymlEncoder.Close()
+	dumpFile.Close()
+
+	msg := fmt.Sprintf(`
+This error should never have happened.
+
+We collected some diagnostic data and saved it in %s.
+
+It will be great if you could raise an issue at https://github.com/turbot/steampipe/issues/new and upload the file.
+
+Note: Please remove any sensitive data from the file before uploading.
```

Comment:
We are listing the postgres processes, which may contain sensitive information.

Reply:
```It would be great if you could raise an issue at https://github.com/turbot/steampipe/issues/new and upload the file.

Note: Please review the file first and confirm you are happy to share the data collected```